root/components/ReleaseGroupsList.js

```diff
@@ -0,0 +1,263 @@
+  +$c: CatalystContextT,
+  +checkboxes?: string,
+  +noGrouping?: boolean,
+  +noRatings?: boolean,
+  +order?: string,
+  +seriesItemNumbers?: {+[number]: string},
+  +showArtists?: boolean,
+  +sortable?: boolean,
+|};
+
+type ReleaseGroupsListEntryProps = {|
+  +$c: CatalystContextT,
+  +checkboxes?: string,
+  +index: number,
+  +noGrouping?: boolean,
+  +noRatings?: boolean,
+  +releaseGroup: ReleaseGroupT,
+  +seriesItemNumbers?: {+[number]: string},
+  +showArtists?: boolean,
+|};
+
+type ReleaseGroupsListProps = {|
+  +$c: CatalystContextT,
+  +checkboxes?: string,
+  +noGrouping?: boolean,
+  +noRatings?: boolean,
+  +order?: string,
+  +releaseGroups: $ReadOnlyArray<ReleaseGroupT>,
+  +seriesItemNumbers?: {+[number]: string},
+  +showArtists?: boolean,
+  +sortable?: boolean,
+|};
+
+const ReleaseGroupsListHeader = ({
+  $c,
+  checkboxes,
+  noGrouping,
+  noRatings,
+  order,
+  seriesItemNumbers,
+  showArtists,
+  sortable,
+}: ReleaseGroupsListHeaderProps) => (
+  <thead>
+    <tr>
+      {$c.user_exists && checkboxes ? (
+        <th style={{width: '1em'}}>
+          <input type="checkbox" />
+        </th>
+      ) : null}
+      {seriesItemNumbers ? <th style={{width: '1em'}}>{l('#')}</th> : null}
+      <th className="year c">
+        {sortable
+          ? (
+            <SortableTableHeader
+              label={l('Year')}
+              name="year"
+              order={order}
+            />
+          )
+          : l('Year')}
+      </th>
+      <th>
+        {sortable
+          ? (
+            <SortableTableHeader
+              label={l('Title')}
+              name="name"
+              order={order}
+            />
+          )
+          : l('Title')}
+      </th>
+      {showArtists ? <th className="artist">{l('Artist')}</th> : null}
+      {noGrouping ? (
+        <th>
+          {sortable
+            ? (
+              <SortableTableHeader
+                label={l('Type')}
+                name="primary_type"
+                order={order}
+              />
+            )
+            : l('Type')}
+        </th>
+      ) : null}
+      {noRatings ? null : <th className="rating c">{l('Rating')}</th>}
+      <th className="count c">{l('Releases')}</th>
+    </tr>
+  </thead>
+);
+
+const ReleaseGroupsListEntry = ({
+  $c,
+  checkboxes,
+  index,
+  noGrouping,
+  noRatings,
+  releaseGroup,
+  seriesItemNumbers,
+  showArtists,
+}: ReleaseGroupsListEntryProps) => (
+  <tr className={loopParity(index)} key={releaseGroup.id}>
+    {$c.user_exists && checkboxes ? (
+      <td>
+        <input
+          name={checkboxes}
+          type="checkbox"
+          value={releaseGroup.id}
+        />
+      </td>
+    ) : null}
+    {seriesItemNumbers ? (
+      <td style={{width: '1em'}}>
+        {seriesItemNumbers[releaseGroup.id]}
+      </td>
+    ) : null}
+    <td className="c">
+      {releaseGroup.firstReleaseDate
+        ? new Date(releaseGroup.firstReleaseDate).getFullYear()
```

Comment:
Since we get date as a string, we can't just do .year like in TT. I think this is the easiest alternative?

Reply:
My preference would be to change the `firstReleaseDate` serialization to use `$self->first_release_date->TO_JSON` rather than `->format`, so that we can use `.year` here. (The places that need it as a string can use the `formatDate` utility.)

Technically the `Date` constructor wouldn't be able to parse our `PartialDate` string format if the dates are actually partial, e.g. `1999-??-01`.

Reply:
The problem is firstReleaseDate is currently used in Autocomplete.js, where it is taken from TO_JSON for direct search, but taken from the WS (as a string) for indexed.

I guess one option would be to serialize both firstReleaseDate and firstReleasePartialDate, and use the latter here but the former for Autocomplete.js?

Reply:
Ah, I forgot about indexed. Then what you have works except I'd import root/static/scripts/common/utility/parseDate.js and do `parseDate(releaseGroup.firstReleaseDate).year` to get the year.